packages/language-service/src/completions.ts

```diff
@@ -531,6 +537,62 @@ export class CompletionBuilder<N extends TmplAstNode|AST> {
     return directive?.tsSymbol;
   }
 
+  private isAnimationCompletions(): this is ElementAnimationCompletionBuilder {
+    return (this.node instanceof TmplAstBoundAttribute &&
+            this.node.type === BindingType.Animation) ||
+        (this.node instanceof TmplAstBoundEvent && this.node.type === ParsedEventType.Animation);
+  }
+
+  private getAnimationCompletions(this: ElementAnimationCompletionBuilder):
+      ts.WithMetadata<ts.CompletionInfo>|undefined {
+    if (this.node instanceof TmplAstBoundAttribute) {
+      const animations = this.compiler.getTemplateTypeChecker()
+                             .getDirectiveMetadata(this.component)
+                             ?.animationTriggerNames?.triggerNames;
+      const replacementSpan = makeReplacementSpanFromParseSourceSpan(this.node.keySpan);
+      if (animations) {
+        const entries = buildAnimationCompletionEntries(
+            [...animations, '.disabled'], replacementSpan, DisplayInfoKind.ATTRIBUTE);
+        return {
+          entries,
+          isGlobalCompletion: false,
+          isMemberCompletion: false,
+          isNewIdentifierLocation: true,
+        };
```

Comment:
The style in the compiler generally follows keeping the happy/common path in as little nesting as possible and using early returns for the other paths:
 
```suggestion
      const animations = this.compiler.getTemplateTypeChecker()
                             .getDirectiveMetadata(this.component)
                             ?.animationTriggerNames?.triggerNames;
      if (!animations) {
        return undefined;
      }
      const replacementSpan = makeReplacementSpanFromParseSourceSpan(this.node.keySpan);
      const entries = buildAnimationCompletionEntries(
            [...animations, '.disabled'], replacementSpan, DisplayInfoKind.ATTRIBUTE);
      return {
          entries,
          isGlobalCompletion: false,
          isMemberCompletion: false,
          isNewIdentifierLocation: true,
      };
```

Same comment for the block below

Reply:
And `if (!animations)` would be `if (animations === undefined)`; at least in the compiler we're explicit about checking for exact nullable values instead of relying on truthy checks.